### DISCUSSION TITLE
Allow restriction of N types of block in Block List

### DISCUSSION THREAD

**Discussion by Bakersbakebread** [umbraco/Umbraco-CMS#14188] [Features and ideas]
Pursuant to a conversation in Discord where I was asking about the best way to achieve this, I thought raising a discussion to see if this feature was a good candidate for being worked on in core. It seems like I wasn't the only one who needed the functionality, or at least an "official" workaround/solution to deliver.

The functionality is essentially allowing a restricted amount of blocks based on block type in a block list.

Example allowed
`Block A` - only 2 allowed in list.
`Block B` - only 1 allowed in list.
`Block C` - infinite amount allowed in list.

As an example, it could be configured when adding the block to the list: 
![image](https://user-images.githubusercontent.com/29239704/235635905-4edc7ee9-96c8-4fb3-aa83-267888d37663.png)

**Comment by Matthew-Wise** [comment]
Id also love to see a way of hiding a block choice via the notifcation handlers. So we can show different block options to different users etc.

**Reply by dennispedersenakqa** [reply]
This is definitely something I would also love to see.

**Comment by tomotrepo** [comment]
Definitely one of the better suggestions. I have searched the forums several times for this but without success. Let me say that [this](https://github.com/rasmusjp/umbraco-embedded-content) editor has it and Rasmus is the author of both: [this](https://github.com/rasmusjp/umbraco-embedded-content) plugin and the new one. In addition, [this](https://github.com/rasmusjp/umbraco-embedded-content) plugin ALREADY has a built-in 'hidden' function. I really miss exactly such an editor.

**Comment by pasindu1997** [comment]
OMG, this is a feature I was also looking for...Do we have a work around for this? @Bakersbakebread
